shared/reducers/settings.js

```diff
@@ -202,6 +202,31 @@ function reducer(state: Types.State = Constants.initialState, action: SettingsGe
         ...state,
         waitingForResponse: waiting,
       }
+    case SettingsGen.unfurlSettingsRefreshed:
+    case SettingsGen.unfurlSettingsSaved:
+      const {mode, whitelist} = action.payload
+      return {
+        ...state,
+        chat: {
+          ...state.chat,
+          unfurl: {
+            unfurlMode: mode,
+            unfurlWhitelist: whitelist,
+            unfurlError: undefined,
+          },
+        },
+      }
+    case SettingsGen.unfurlSettingsError:
+      return {
```

Comment:
Ah, I see the settings store isn't using immutable -- if it was, it'd be cool to be able to use state.merge() to avoid all these `...`s.

Reply:
Yeah, I just mimicked what other reducers were doing here. I can see the advantage of using something like Immutable, if the shape of this store was more complicated it seems easy to mess this up.